packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/rest_api/FlutterRestApi.kt

```diff
@@ -0,0 +1,252 @@
+            val apiName: String? = inputs.getApiPath()
+            val options: RestOptions = FlutterRestInputs(request).getRestOptions()
+
+            try {
+
+                if (apiName == null) {
+                    var operation: RestOperation? = function3(options,
+                            Consumer { result -> prepareRestResponseResult(flutterResult, result, methodName, cancelToken) },
+                            Consumer { error ->
+                                prepareError(
+                                        flutterResult,
+                                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                                        cancelToken,
+                                        error)
+                            }
+                    )
+                    OperationsManager.addOperation(cancelToken, operation!!)
+                } else {
+                    var operation: RestOperation? = function4(
+                            apiName,
+                            options,
+                            Consumer { result -> prepareRestResponseResult(flutterResult, result, methodName, cancelToken) },
+                            Consumer { error ->
+                                prepareError(
+                                        flutterResult,
+                                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                                        cancelToken,
+                                        error)
+                            }
+                    )
+                    OperationsManager.addOperation(cancelToken, operation!!)
+                }
+
+            } catch (e: Exception) {
+                prepareError(
+                        flutterResult,
+                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                        cancelToken,
+                        e)
+            }
+        }
+
+        fun prepareError(flutterResult: Result, msg: String, cancelToken: String, error: Exception) {
+            if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+            FlutterApiError.postFlutterError(flutterResult, msg, error)
+        }
+
+        private fun prepareRestResponseResult(flutterResult: Result, result: RestResponse, methodName: String, cancelToken: String = "") {
+
+            var restResponse = FlutterRestResponse(result)
+
+            if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+
+            // if code is not 200 then throw an exception
+            if (!result.code.isSuccessful) {
+                FlutterApiError.handleAPIError(
+                        flutterResult,
+                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                        "The HTTP response status code is [" + result.code.toString().substring(16, 19) + "].",
```

Comment:
Let's add a comment here about what result.code actually is which will be helpful to provide context while parsing the native response